### ISSUE TITLE
Trailing Comma - Snowflake

### ISSUE THREAD

**Issue by ryaminal** [DerekStride/tree-sitter-sql#244]
Curious what the thoughts are on supporting a trailing comma in SQL? Snowflake just announced [support](https://docs.snowflake.com/en/release-notes/2024/8_11#select-supports-trailing-commas) for this in their dialect.
Example:
```sql
SELECT emp_id,
       name,
       dept,
FROM employees;
```
This doesn't appear to be handled by `tree-sitter-sql` yet(as it was just announced) but curious if this is something tree-sitter-sql wants to support?

**Comment by DerekStride** [comment]
I'm up for adding it! You interested in writing the PR?

**Comment by ryaminal** [comment]
yeah, i'll give it a go. do you want it behind a flag or anything?

**Comment by DerekStride** [comment]
Adding it to the existing behaviour is good

**Comment by matthias-Q** [comment]
Not sure if I am a fan of adding the trailing comma. This is a convenience addition by snowflake and - as of now - is only supported by snowflake. It saved me more than once to see an error in the highlighting, when having a trailing comma that my query engines (postgres/impala/athena) do not accept. Since this parser does not distinguish between dialects it would not through an error.

In this particular case it would not be wise to be permissive.

**Comment by matthias-Q** [comment]
Oh, you are right. Nice find. I was sure it was given an error. Not sure if that got lost in some commit.
